### ISSUE TITLE
Error with on-demand-entries-ping: Unexpected token i in JSON at position 0

### ISSUE THREAD

**Issue by rohmanhm** [zeit/next.js#1667]
```javascript
app.get('/:username', handle.profile)
app.get('/:username/:id', (req, res) => {
  res.send('id')
})
```
Got error when I define express route like above.
![screen shot 2017-04-08 at 16 38 00](https://cloud.githubusercontent.com/assets/7524911/24827725/c4c5e05c-1c79-11e7-9f67-079d9911738d.png)

But all be fine if I define my route like this 
```javascript
app.get('/:username', handle.profile)
app.get('/:username/forum/:id', (req, res) => {
  res.send('id')
})
```
Any solution?

**Comment by pungggi** [comment]
see #1596

**Comment by rohmanhm** [comment]
That was different cases.
In my case, the error throws from the route inside my custom express server

**Comment by rohmanhm** [comment]
I ask on **StackOverflow** with more detailed information http://stackoverflow.com/questions/43301797/nextjs-expressjs-error-with-on-demand-entries-ping

**Comment by rohmanhm** [comment]
I wonder with this line https://github.com/zeit/next.js/blob/master/server/on-demand-entry-handler.js#L107

**Comment by rohmanhm** [comment]
I have tried with KoaJS too, and still get error
> Error with on-demand-entries-ping: Unexpected token i in JSON at position 0 

```javascript
router.get('/:username/:id', async ctx => {
    console.log(ctx.params)
    await app.render(ctx.req, ctx.res, '/a', ctx.query)
    ctx.respond = false
})
```

**Comment by arunoda** [comment]
This should be fixed by the latest version: `2.1.1`.
The PR is: https://github.com/zeit/next.js/pull/1658
